### PR TITLE
Refactor resource call

### PR DESCRIPTION
**Pull request by ZhangMan1126**
refactor the resource-based solution
gene bindings within resource handing or dependent resources handing
add option to skip failed sql commands in the generated tests

### CODE REVIEW

File: client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/operations/InsertionResultsDto.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class InsertionResultsDto {
+
+    /**
+     * primary key generated with the insertions
+     */
```
Comment (arcuri82) [inline client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/operations/InsertionResultsDto.java:12]: specify exactly what are the `key` and `value` in this map

File: client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/operations/InsertionResultsDto.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    public Map<Long, Long> idMapping = new HashMap<>();
+
+    /**
+     * whether the insertion at the index executed successfully
+     */
+    public List<Boolean> executionResults = new ArrayList<>();
```
Comment (arcuri82) [inline client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/operations/InsertionResultsDto.java:18]: specify that the list is ordered by index... of what exactly? the actions in the tests? or a specific series of SQL commands?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/org/evomaster/core/database/DbAction.kt [hunk 3]
```diff
@@ -52,6 +52,10 @@ class DbAction(
         }
     }
 
+    /**
+     * whether the insertion was executed successfully
+     */
+    var insertExecutedSuccessfully = false
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/database/DbAction.kt:58]: didn't we discuss that this should not be in the action itself, but on the result? 
or was this supposed to be refactored at a later stage (but, in this, it misses a TODO comment)

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/org/evomaster/core/database/DbAction.kt (hunk 3 above)
Comment (ZhangMan1126) [reply core/src/main/kotlin/org/evomaster/core/database/DbAction.kt:59]: I will refactor this later. now the TODO is added.

**Pull request merged**